kornia/geometry/linalg.py

```diff
@@ -213,6 +214,53 @@ def transform_points(trans_01: torch.Tensor,
     return points_0
 
 
+def transform_boxes(trans_mat: torch.Tensor, boxes: torch.Tensor, mode: str = "xyxy") -> torch.Tensor:
+
+    r""" Function that applies a transformation matrix to a box or batch of boxes. Boxes must
+    be a tensor of the shape (4, ) or a batch of boxes (B, 4) and trans_mat must be a (3, 3)
+    transformation matrix
+
+    Args:
+        trans_mat (torch.Tensor): The transformation matrix to be applied
+        boxes (torch.Tensor): The boxes to be transformed
+        mode (str): The format in which the boxes are provided. If set to 'xyxy' the boxes
+                    are assumed to be in the format (xmin, ymin, xmax, ymax). If set to 'xywh'
+                    the boxes are assumed to be in the format (xmin, ymin, width, height).
+                    Default: 'xyxy'
+    Returns:
+        torch.Tensor: The set of transformed points in the specified mode
+
+
+    """
+
+    if not torch.is_tensor(boxes):
+        raise TypeError(f"Boxes type is not a torch.Tensor. Got {type(boxes)}")
+
+    if not torch.is_tensor(trans_mat):
+        raise TypeError(f"Tranformation matrix type is not a torch.Tensor. Got {type(trans_mat)}")
+
+    if not isinstance(mode, str):
+        raise TypeError(f"Mode must be a string. Got {type(mode)}")
+
+    if mode not in ("xyxy", "xywh"):
+        raise ValueError(f"Mode must be one of 'xyxy', 'xywh'. Got {mode}")
+
+    boxes = boxes.unsqueeze(0)
```

Comment:
What's the boxes shape then ? I would assume Nx4 or BxNx4

Reply:
Agree, I would ask for Nx4 or BxNx4 as you usually have several boxes per image.